cli/running/lua/play.lua

```diff
@@ -0,0 +1,192 @@
+           (spaces and (sid == nil or not find_in_list(sid, spaces))) or
+           (replicas and not find_in_list(rid, replicas)) then
+            -- Pass this tuple, luacheck: ignore.
+        else
+            cb(record)
+        end
+    end
+end
+
+local function play()
+    local opts = keyword_arguments
+    local uri = table.remove(positional_arguments, 1)
+    if uri == nil then
+        log.error("Internal error: empty URI is provided")
+        os.exit(1)
+    end
+    local remote = netbox.new(uri)
+    if not remote:wait_connected() then
+        log.error("Fatal error: no connection to the host '%s'", uri)
+        os.exit(1)
+    end
+    for _, file in ipairs(positional_arguments) do
+        print(string.format("• Play is processing file '%s' •", file))
+        local gen, param, state = xlog.pairs(file)
+        filter_xlog(gen, param, state, opts, function(record)
+            local sid = record.BODY and record.BODY.space_id
+            if sid ~= nil then
+                local args, so = {}, remote.space[sid]
+                if so == nil then
+                    log.error("Fatal error: no space #%s, stopping work", sid)
+                    os.exit(1)
+                end
+                table.insert(args, so)
+                table.insert(args, record.BODY.key)
+                table.insert(args, record.BODY.tuple)
+                table.insert(args, record.BODY.operations)
+                so[record.HEADER.type:lower()](unpack(args))
+            end
+        end)
+        print(string.format("• Done with file '%s' •", file))
+    end
+    print("\n• Play result: completed successfully •")
+    remote:close()
+end
+
+local function populate_arguments()
+    local function keyword_arguments_populate(ka)
+        ka                = ka                or {}
+        ka.from           = ka.from           or 0
+        ka.to             = ka.to             or -1ULL
+        ka['show-system'] = ka['show-system'] or false
+        ka.format         = ka.format         or 'yaml'
+        return ka
+    end
+
+    -- Returns the command name, file list and named parameters.
+    local function parameters_parse(parameters)
+        local positional_arguments, keyword_arguments = {}, {}
+        for key, val in pairs(parameters) do
+            if type(key) == 'number' then
+                positional_arguments[key] = val
+            else
+                keyword_arguments[key] = val
+            end
+        end
+        return positional_arguments, keyword_arguments
+    end
+
+    local parameters = argparse(arg, {
+        { 'space',       'number+' },
+        { 'show-system', 'boolean' },
+        { 'from',        'number'  },
+        { 'to',          'number'  },
+        { 'help',        'boolean' },
+        { 'format',      'string'  },
+        { 'replica',     'number+' },
+        { 'language',    'string'  },
+    })
+
+    positional_arguments, keyword_arguments = parameters_parse(parameters)
```

Comment:
No need to work through global variables here.

Reply:
Done, now passed as arguments:

`play(positional_arguments, keyword_arguments)`